### PR TITLE
Sagemaker integration

### PR DESCRIPTION
**Pull request by gianlucadetommaso**
Faciliate running of Fortuna on Amazon SageMaker by:

- Implement script to build and push docker image.
- Implement function that runs a training job on Amazon Sagemaker from a configuration yaml file. Configuration includes 
the entrypoint, AWS info, hyperparameters, etc.

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe):

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -165,6 +166,38 @@ Several usage examples are found in the
...
+============================
+We offer a simple pipeline that allows you to run Fortuna on Amazon SageMaker with minimal effort.
+
+1. Create an AWS account - it is free! Store the account ID and the region where you want to launch training jobs.
+
+2. First, `update your local AWS credentials <https://docs.aws.amazon.com/cli/latest/userguide/cli-authentication-short-term.html>`__.
+   Then you need to build and `push a Docker image to an Amazon ECR repository <https://docs.aws.amazon.com/AmazonECR/latest/userguide/docker-push-ecr-image.html>`__.
+   This `script <https://github.com/awslabs/fortuna/tree/main/fortuna/docker/build_and_push.sh>`_ will help you doing so -
+   it will require your AWS account ID and region. If you need other packages to be included in your Docker image,
+   you should consider customize the `Dockerfile <https://github.com/awslabs/fortuna/tree/main/fortuna/docker/Dockerfile>`_.
```
Comment (wesk) [inline README.rst:179]:
```suggestion
   you should consider customizing the `Dockerfile <https://github.com/awslabs/fortuna/tree/main/fortuna/docker/Dockerfile>`_.
```

File: benchmarks/transformers/prob_model_text_classification.py
Comment (wesk) [inline benchmarks/transformers/prob_model_text_classification.py:1]: This file was already merged as part of a separate PR adding the example, right? Should it be in this PR?

File: fortuna/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+RUN apt update && apt install -y python3-pip
+
+RUN ln -sf /usr/bin/python3 /usr/bin/python && \
+ln -sf /usr/bin/pip3 /usr/bin/pip
+
+RUN pip --no-cache-dir install --upgrade pip setuptools_rust
+
+# Install ML Packages built with CUDA11 support
+RUN ln -s /usr/lib/cuda /usr/local/cuda-11.8
+RUN pip --no-cache-dir install --upgrade "jax[cuda11_local]" -f https://storage.googleapis.com/jax-releases/jax_cuda_releases.html
```
Comment (wesk) [inline fortuna/docker/Dockerfile:16]: Is it not available as a regular pip package?

File: fortuna/docker/build_and_push.sh [hunk 3]
```diff
@@ -0,0 +1,133 @@
+#!/bin/bash
+
+helpFunction()
```
Comment (wesk) [inline fortuna/docker/build_and_push.sh:3]: Nice

File: benchmarks/transformers/prob_model_text_classification.py
Comment (gianlucadetommaso) [reply benchmarks/transformers/prob_model_text_classification.py:1]: Yes, I just renamed it.

File: fortuna/docker/Dockerfile (hunk 2 above)
Comment (gianlucadetommaso) [reply fortuna/docker/Dockerfile:16]: I let users install this outside Fortuna because the CUDA version they need depends on the machine.

**Pull request merged**